corehq/apps/app_manager/views/utils.py

```diff
@@ -167,7 +167,22 @@ def overwrite_app(app, master_build, report_map=None):
     ])
     master_json = master_build.to_json()
     app_json = app.to_json()
-    form_ids_by_xmlns = _get_form_ids_by_xmlns(app_json)  # do this before we change the source
+
+    # Corresponding forms in a master app and linked app need to have the same XMLNS but different unique ids,
+    # so the linked app needs to know if there are any new forms and, if so, assign those forms new unique ids.
+    # To do this lookup, get the XMLNSes from the the most recent versions of this app pulled from each master
+    # and compare those to the XMLNSes present in this app.
+    form_ids_by_xmlns = {}
+    master_app_briefs = app.get_master_app_briefs()
+    for brief in master_app_briefs:
```

Comment:
does the order here matter?

Reply:
It shouldn't, since all versions of the linked app should be using the same unique id for the same XMLNS.